packages/minimongo/minimongo.js

```diff
@@ -782,8 +782,22 @@ LocalCollection.prototype.update = function (selector, mod, options, callback) {
   // generate an id for it.
   var insertedId;
   if (updateCount === 0 && options.upsert) {
-    var newDoc = LocalCollection._removeDollarOperators(selector);
+    var newDoc = {};
+     
+    if (LocalCollection._selectorIsId(selector)) {
+      selector = {_id: selector}
```

Comment:
Can we avoid re-assigning to the `selector`, and instead use a new variable to represent the modified selector?